common4j/src/main/com/microsoft/identity/common/java/telemetry/Telemetry.java

```diff
@@ -134,7 +136,8 @@ public synchronized void addObserver(@SuppressWarnings(WarningType.rawtype_warni
             mObservers = new ConcurrentLinkedQueue<>();
         }
 
-        mObservers.add(observer);
+        if (!mObservers.contains(observer))
+            mObservers.add(observer);
```

Comment:
We should see if we can safely change `mObservers` to be a `ConcurrentHashMap` as opposed to a `ConcurrentLinkedQueue`. If we do that then the above two lines just become a call to the `putIfAbsent` method

Reply:
We don't necessarily have to do this right now. It can be done in a separate PR